src/qurveros/optspacecurve.py

```diff
@@ -50,21 +51,26 @@ def __init__(self, *, curve, order, interval, params=None,
         self.params_history = None
         self.loss_grad = lambda params: None
 
+        #if params is not None:
+         #   params = self._cast_params_to_float(params)
         super().__init__(curve=curve,
                          order=order,
                          interval=interval,
                          params=params,
                          deriv_array_fun=deriv_array_fun)
+    
+    def set_params(self, params):
```

Comment:
This overridden `set_params` function is not necessary because it does exactly the same thing as the parent class function.